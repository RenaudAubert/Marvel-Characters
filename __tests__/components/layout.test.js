import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import apiResult from '../__mocks__/characters';

// Component to be tested
import {
  FavButton, FavElement, Footer, Header
} from '../../src/client/layout';

const characters = apiResult.data.results;

describe('<FavButton />', () => {
  const mockOnClick = jest.fn();
  const wrapper = shallow(<FavButton characterId={1} onFavClicked={mockOnClick} />);

  describe('return()', () => {
    test('renders the FavButton component', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('onClick()', () => {
    test('successfully calls the onClick handler', () => {
      wrapper.find('button').simulate('click');
      expect(mockOnClick.mock.calls.length).toEqual(1);
    });
  });
});

describe('<FavElement />', () => {
  const [character] = characters;
  const mockOnClick = jest.fn();
  const wrapper = shallow(
    <FavElement id={character.id} name={character.name} onDeleteFav={mockOnClick} />
  );

  describe('return()', () => {
    test('renders the FavElement component', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('onClick()', () => {
    test('successfully calls the onClick handler', () => {
      wrapper.find('button').simulate('click');
      expect(mockOnClick.mock.calls.length).toEqual(1);
    });
  });
});

describe('<Footer />', () => {
  describe('return()', () => {
    test('renders the Footer component with props', () => {
      const wrapper = shallow(<Footer attributionText="test" />);
      expect(wrapper.contains(<div className="container">test</div>)).toEqual(true);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders the Footer component w/o props', () => {
      const wrapper = shallow(<Footer />);
      expect(wrapper.find('.container').children().exists()).toEqual(false);
    });
  });
});

describe('<Header />', () => {
  const mockOnClick = jest.fn();
  describe('return()', () => {
    test('renders the Header component with props', () => {
      const wrapper = shallow(<Header favCharacters={characters} onDeleteFav={mockOnClick} />);
      expect(wrapper.find('.dropdown-menu').children()).toHaveLength(characters.length);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders the Header component w/o props', () => {
      const wrapper = shallow(<Header onDeleteFav={mockOnClick} />);
      expect(wrapper.contains(<p className="text-center">No favorite characters</p>)).toEqual(true);
    });
  });
});
