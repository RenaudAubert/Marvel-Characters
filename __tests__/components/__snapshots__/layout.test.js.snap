// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavButton /> return() renders the FavButton component 1`] = `
<button
  className="btn btn-sm fa-button"
  onClick={[Function]}
  type="button"
>
  <i
    className="far fa-star text-warning fa-lg"
  />
</button>
`;

exports[`<FavElement /> return() renders the FavElement component 1`] = `
<div
  aria-label="Favorite List"
  className="btn-group"
  role="group"
>
  <Link
    className="dropdown-item btn btn-secondary text-center"
    replace={false}
    to="/1011334"
  >
    3-D Man
  </Link>
  <button
    className="close btn btn-secondary"
    onClick={[Function]}
    type="button"
  >
    <i
      aria-hidden="true"
      className="fa fa-trash"
    />
  </button>
</div>
`;

exports[`<Footer /> return() renders the Footer component with props 1`] = `
<footer
  className="footer text-muted"
>
  <div
    className="container"
  >
    test
  </div>
</footer>
`;

exports[`<Header /> return() renders the Header component with props 1`] = `
<header>
  <div
    className="navbar navbar-dark bg-dark box-shadow"
  >
    <div
      className="container d-flex justify-content-between"
    >
      <Link
        className="navbar-brand d-flex align-items-center"
        replace={false}
        to="/"
      >
        Marvel Heroes
      </Link>
      <div
        className="dropdown"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          className="btn btn-secondary"
          data-toggle="dropdown"
          id="dropdownFavorite"
          type="button"
        >
          My Favorites
        </button>
        <div
          aria-labelledby="dropdownMenuButton"
          className="dropdown-menu"
        >
          <FavElement
            id={1011334}
            key="1011334"
            name="3-D Man"
            onDeleteFav={[MockFunction]}
          />
        </div>
      </div>
    </div>
  </div>
</header>
`;
